### DISCUSSION TITLE
Sticky Loadbalancing

### DISCUSSION THREAD

**Discussion by jcbjoe** [emmett-framework/granian#352] [Q&A]
Hi!

Is there anyway to use cookies to specify which worker to route specific requests to? The reason this is important for us is because we store state in a local dictionary which is much faster to access than going to Redis or Mongo. 

Currently, I'm running 4 separate instances of Granian and use HAProxy for sticky routing. However some of the Granian processes seem to be running on the same cores.

Thanks!
Joe

**Comment by gi0baro** [comment]
> Is there anyway to use cookies to specify which worker to route specific requests to? The reason this is important for us is because we store state in a local dictionary which is much faster to access than going to Redis or Mongo.
> 
> Currently, I'm running 4 separate instances of Granian and use HAProxy for sticky routing. 

No, there's currently no support for that. Using a proxy server like HAProxy or similar is the correct way to go for this.

> However some of the Granian processes seem to be running on the same cores.

Granian runs 2 different loops (at least in ASGI and RSGI protocols), so you might want to run NCpu/2 workers.

**Reply by jcbjoe** [reply]
Hi @gi0baro,

Thanks for the update, Is there any plans to implement sticky load balancing?

Also is there any downsides to running 4 separate instances of Granian on different ports?

Thanks,
Joe

**Reply by gi0baro** [reply]
> Is there any plans to implement sticky load balancing?

No, it feels a bit out of context for the project at the moment.

> Also is there any downsides to running 4 separate instances of Granian on different ports?

Not that I'm aware of.
